.github/workflows/cloud-deploy-embed.yml

```diff
@@ -0,0 +1,51 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Deploy EMBED
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  workflow_call:
+    inputs:
+      env:
+        description: 'Deployment environment - choose between dev and prod'
```

Comment:
```suggestion
        description: 'Deployment environment - choose between staging and production'
```

Reply:
@AliaksandrRyzhou how about adding three options here, US/EU and Staging, considering each environment differently, this will make the task definition simpler as we will avoid the if conditions for every environment, later on.

As for the values per environment such as the widget_api_url we can fetch them from a predefined map of such values.

Reply:
If we create the three options here it means that in `Production` deployment we have to call twice `Deploy EMBED WF` in `cloud-deploy.yml` because the `prod` has two regions which in turn means that we will have to add extra jobs. In this WF we use conditions only to create a few jobs instead of one in the production deployment case, because to make matrix element conditionals won't look greate. All this  jobs are equal for both environments beacause in the end they use the same `cloud-deploy-reusable-embed-deploy.yml` only with different variables. I see what you mean to create something like here:
```
name: Deploy EMBED

on:
  workflow_call:
    inputs:
      env:
        description: 'Deployment environment - choose between staging, production-eu, and production-us'
        required: true
        type: string
        default: 'staging'

jobs:
  deploy_embed:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        environment: [staging, production-eu, production-us]

    steps:
      - name: Set Environment Variables
        id: set-env-vars
        run: |
          if [ "${{ matrix.environment }}" == "staging" ]; then
            echo "env=Development" >> $GITHUB_ENV
            echo "widget_url=https://dev.widget.novu.co" >> $GITHUB_ENV
            echo "netlify_deploy_message=Dev deployment" >> $GITHUB_ENV
            echo "netlify_alias=dev" >> $GITHUB_ENV
            echo "netlify_gh_env=development" >> $GITHUB_ENV
            echo "netlify_site_id=22682666-3a8d-40be-af26-017bfadf5ae9" >> $GITHUB_ENV
          elif [ "${{ matrix.environment }}" == "production-eu" ]; then
            echo "env=Production" >> $GITHUB_ENV
            echo "widget_url=https://eu.widget.novu.co" >> $GITHUB_ENV
            echo "netlify_deploy_message=Production deployment" >> $GITHUB_ENV
            echo "netlify_alias=prod" >> $GITHUB_ENV
            echo "netlify_gh_env=Production" >> $GITHUB_ENV
            echo "netlify_site_id=0c830b50-df83-480b-ba36-a7f3176efcc8" >> $GITHUB_ENV
          elif [ "${{ matrix.environment }}" == "production-us" ]; then
            echo "env=Production" >> $GITHUB_ENV
            echo "widget_url=https://widget.novu.co" >> $GITHUB_ENV
            echo "netlify_deploy_message=Production deployment" >> $GITHUB_ENV
            echo "netlify_alias=prod" >> $GITHUB_ENV
            echo "netlify_gh_env=Production" >> $GITHUB_ENV
            echo "netlify_site_id=0689c015-fca0-4940-a26d-3e33f561bc48" >> $GITHUB_ENV
          fi

      - name: Deploy
        uses: ./.github/workflows/cloud-deploy-reusable-embed-deploy.yml
        with:
          environment: ${{ env.env }}
          widget_url: ${{ env.widget_url }}
          netlify_deploy_message: ${{ env.netlify_deploy_message }}
          netlify_alias: ${{ env.netlify_alias }}
          netlify_gh_env: ${{ env.netlify_gh_env }}
          netlify_site_id: ${{ env.netlify_site_id }}
        secrets: inherit

```
But as you can see and as I said we we should call it a few times. We will receive the same difference between environments on a level of the workflows higher. @SokratisVidros Please let me know your opinion.

Reply:
For posterity, this was resolved offline.